components/layout/webrender_helpers.rs

```diff
@@ -359,19 +351,22 @@ impl WebRenderDisplayListConverter for DisplayList {
                             scroll_layer_id: Option<webrender_traits::ScrollLayerId>,
                             frame_builder: &mut WebRenderFrameBuilder)
                             -> webrender_traits::StackingContextId {
-        let mut traversal = DisplayListTraversal {
-            display_list: self,
-            current_item_index: 0,
-            last_item_index: self.list.len() - 1,
-        };
+        let mut traversal = DisplayListTraversal::new(self);
+        let item = traversal.next();
+        match item {
+            Some(&DisplayItem::PushStackingContextClass(ref stacking_context_item)) => {
+                let stacking_context = &stacking_context_item.stacking_context;
+                stacking_context.convert_to_webrender(&mut traversal,
+                                                      api,
+                                                      pipeline_id,
+                                                      epoch,
+                                                      scroll_layer_id,
+                                                      ScrollPolicy::Scrollable,
+                                                      frame_builder)
+            }
+            _ => panic!("DisplayList did not start with StackingContext."),
```

Comment:
nit: `unreachable!("...")` is a little more semantically meaningful than `panic!`